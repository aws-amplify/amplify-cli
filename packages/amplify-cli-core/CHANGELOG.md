# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 1.1.0 (2020-07-23)


### Features

* feature flag implementation ([#4891](https://github.com/aws-amplify/amplify-cli/issues/4891)) ([4639450](https://github.com/aws-amplify/amplify-cli/commit/463945029cfe861f74986d9a8b9af6b827d2063d))
