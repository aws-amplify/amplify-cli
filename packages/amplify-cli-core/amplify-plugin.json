{
    "name": "core",
    "type": "core",
    "commands": [
        "plugin",
        "diagnose",
        "version"
    ],
    "subscriptions":[]
}