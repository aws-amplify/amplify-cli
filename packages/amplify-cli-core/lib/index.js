"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExeInfo = void 0;
const ExeInfo = __importStar(require("./exeInfo"));
exports.ExeInfo = ExeInfo;
__exportStar(require("./banner-message"), exports);
__exportStar(require("./category-interfaces"), exports);
__exportStar(require("./cfnUtilities"), exports);
__exportStar(require("./cliConstants"), exports);
__exportStar(require("./cliContext"), exports);
__exportStar(require("./cliContextEnvironmentProvider"), exports);
__exportStar(require("./cliEnvironmentProvider"), exports);
__exportStar(require("./cliGetCategories"), exports);
__exportStar(require("./cliRemoveResourcePrompt"), exports);
__exportStar(require("./cliViewAPI"), exports);
__exportStar(require("./customPoliciesUtils"), exports);
__exportStar(require("./deploymentSecretsHelper"), exports);
__exportStar(require("./deploymentState"), exports);
__exportStar(require("./errors"), exports);
__exportStar(require("./exitOnNextTick"), exports);
__exportStar(require("./feature-flags"), exports);
__exportStar(require("./hooks"), exports);
__exportStar(require("./isCI"), exports);
__exportStar(require("./isPackaged"), exports);
__exportStar(require("./jsonUtilities"), exports);
__exportStar(require("./overrides-manager"), exports);
__exportStar(require("./permissionsBoundaryState"), exports);
__exportStar(require("./plugin-facade"), exports);
__exportStar(require("./serviceSelection"), exports);
__exportStar(require("./spinner"), exports);
__exportStar(require("./state-manager"), exports);
__exportStar(require("./tags"), exports);
__exportStar(require("./types"), exports);
__exportStar(require("./utils"), exports);
__exportStar(require("./errors/amplify-error"), exports);
__exportStar(require("./errors/amplify-fault"), exports);
__exportStar(require("./errors/amplify-exception"), exports);
__exportStar(require("./errors/project-not-initialized-error"), exports);
__exportStar(require("./context/context-extensions"), exports);
__exportStar(require("./context/plugin-platform"), exports);
__exportStar(require("./context/amplify-event"), exports);
__exportStar(require("./context/plugin-collection"), exports);
__exportStar(require("./context/plugin-info"), exports);
__exportStar(require("./context/plugin-manifest"), exports);
__exportStar(require("./context/plugin-verification-result"), exports);
__exportStar(require("./constants"), exports);
__exportStar(require("./help"), exports);
__exportStar(require("./amplify-node-pkg-detector"), exports);
//# sourceMappingURL=index.js.map