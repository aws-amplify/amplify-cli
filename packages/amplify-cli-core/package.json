{
  "name": "amplify-cli-core",
  "version": "1.24.1",
  "description": "Amplify CLI Core",
  "repository": {
    "type": "git",
    "url": "https://github.com/aws-amplify/amplify-cli.git",
    "directory": "packages/amplify-cli-core"
  },
  "author": "Amazon Web Services",
  "license": "Apache-2.0",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "keywords": [
    "graphql",
    "appsync",
    "aws"
  ],
  "scripts": {
    "test": "jest",
    "test-ci": "jest --ci -i",
    "build": "tsc",
    "watch": "tsc -w",
    "clean": "rimraf lib tsconfig.tsbuildinfo"
  },
  "dependencies": {
    "ajv": "^6.12.3",
    "amplify-cli-logger": "1.1.0",
    "ci-info": "^2.0.0",
    "cloudform-types": "^4.2.0",
    "dotenv": "^8.2.0",
    "folder-hash": "^4.0.1",
    "fs-extra": "^8.1.0",
    "globby": "^11.0.3",
    "hjson": "^3.2.1",
    "js-yaml": "^4.0.0",
    "lodash": "^4.17.21",
    "node-fetch": "^2.6.1",
    "open": "^7.3.1",
    "proxy-agent": "^4.0.1",
    "semver": "^7.3.5",
    "which": "^2.0.2"
  },
  "devDependencies": {
    "@types/folder-hash": "^4.0.0",
    "@types/fs-extra": "^8.0.1",
    "@types/hjson": "^2.4.2",
    "@types/json-schema": "^7.0.5",
    "@types/lodash": "^4.14.149",
    "@types/node": "^12.12.6",
    "@types/rimraf": "^3.0.0",
    "@types/uuid": "^8.0.0",
    "amplify-function-plugin-interface": "1.9.0",
    "nock": "^13.0.11",
    "rimraf": "^3.0.0"
  },
  "jest": {
    "transform": {
      "^.+\\.tsx?$": "ts-jest"
    },
    "testRegex": "(src/__tests__/.*.test.ts)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx",
      "json",
      "node"
    ],
    "collectCoverage": true
  }
}
