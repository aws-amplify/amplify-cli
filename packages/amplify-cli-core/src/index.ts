import { ViewResourceTableParams } from './cliViewAPI';
import { ServiceSelection } from './serviceSelection';

export * from './cfnUtilities';
export * from './cliContext';
export * from './cliContextEnvironmentProvider';
export * from './cliEnvironmentProvider';
export * from './feature-flags';
export * from './permissionsBoundaryState';
export * from './jsonUtilities';
export * from './jsonValidationError';
export * from './serviceSelection';
export * from './state-manager';
export * from './tags';
export * from './errors';
export * from './exitOnNextTick';
export * from './isPackaged';
export * from './isCI';
export * from './cliConstants';
export * from './deploymentSecretsHelper';
export * from './deploymentState';
export * from './utils';
export * from './banner-message';
export * from './cliGetCategories';
export * from './cliRemoveResourcePrompt';
export * from './cliViewAPI';
export * from './overrides-manager';
export * from './hooks';
export * from './cliConstants';
export * from './category-interfaces';
export * from './customPoliciesUtils';
export * from './utils/doc-links';
export * from './utils/gql-transformer-version';

// Temporary types until we can finish full type definition across the whole CLI

// Use it for all 'any's where we can't define the type, but doing a strict TypeScript conversion
export type $TSAny = any;

// Use it for all CLI Context class references, it enables a quick way to see what we have on the context
export type $TSContext = {
  amplify: AmplifyToolkit;
  /**
   * @deprecated Use printer from package amplify-prompts instead
   */
  print: IContextPrint;
  migrationInfo: $TSAny;
  projectHasMobileHubResources: boolean;
  prompt: $TSAny;
  exeInfo: $TSAny;
  input: $TSAny;
  parameters: $TSAny;
  usageData: $TSAny;
  runtime: $TSAny;
  pluginPlatform: IPluginPlatform;
  newUserInfo?: $TSAny;
  filesystem: IContextFilesystem;
  template: IContextTemplate;
  updatingAuth: $TSAny;
};

export type CategoryName = string;
export type ResourceName = string;

export type IContextPrint = {
  /**
   * @deprecated Use printer.info from amplify-prompts instead
   */
  info: (message: string) => void;
  /**
   * @deprecated Why are you using this? If you really need it, implement it in amplify-prompts printer.ts
   */
  fancy: (message?: string) => void;
  /**
   * @deprecated Use printer.warn from amplify-prompts instead
   */
  warning: (message: string) => void;
  /**
   * @deprecated Use printer.error from amplify-prompts instead
   */
  error: (message: string) => void;
  /**
   * @deprecated Use printer.success from amplify-prompts instead
   */
  success: (message: string) => void;
  /**
   * @deprecated The next time we refactor code that uses this function, refactor the table function into formatter.ts from amplify-prompts and use that instead
   */
  table: (data: string[][], options?: { format?: 'markdown' | 'lean' }) => void;
  /**
   * @deprecated Use printer.debug from amplify-prompts instead
   */
  debug: (message: string) => void;
  /**
   * @deprecated Use printer.info from amplify-prompts and specify color
   */
  green: (message: string) => void;
  /**
   * @deprecated Use printer.info from amplify-prompts and specify color
   */
  yellow: (message: string) => void;
  /**
   * @deprecated Use printer.info from amplify-prompts and specify color
   */
  red: (message: string) => void;
  /**
   * @deprecated Use printer.info from amplify-prompts and specify color
   */
  blue: (message: string) => void;
};

export type IContextFilesystem = {
  remove: (targetPath: string) => void;
  read: (targetPath: string, encoding?: string) => $TSAny;
  write: (targetPath: string, data: unknown) => void;
  exists: (targetPath: string) => boolean;
  isFile: (targetPath: string) => boolean;
  path: (...pathParts: string[]) => string;
};

export type IContextTemplate = {
  generate: (opts: { template: string; target: string; props: object; directory: string }) => string;
};

export type IPluginPlatform = {
  pluginDirectories: string[];
  pluginPrefixes: string[];
  userAddedLocations: string[];
  lastScanTime: Date;
  maxScanIntervalInSeconds: Number;
  plugins: IPluginCollection;
  excluded: IPluginCollection;
};

export type IPluginCollection = {
  [pluginType: string]: IPluginInfo[];
};

export type IPluginInfo = {
  packageName: string;
  packageVersion: string;
  packageLocation: string;
  manifest: $IPluginManifest;
};

export type DeploymentSecrets = {
  appSecrets: Array<{
    rootStackId: string;

    environments: { [env: string]: { [category: string]: { [resourceName: string]: { [key: string]: string } } } };
  }>;
};

/**
 * Plugins or other packages bundled with the CLI that pass a file to a system command or execute a binary file must export a function named
 * "getPackageAssetPaths" of this type.
 *
 * The function must return the relative paths of all files and folders that the package passes into a system command.
 * If the package is an Amplify plugin, the path must be relative to the location of the amplify-plugin.json file
 * If the package is not an Amplify plugin, the path must be relative to the location of require.resolve('your-package')
 *
 * This function will be executed by the CLI during installation.
 *
 * At runtime, the assets can be retrieved at path.join(pathManager.getAmplifyPackageLibDirPath(packageName), relativePath)
 * where "pathManager" is the PathManager instance exported by this package,
 * "packageName" is the name of your package (used as a key to locate the assets),
 * and "relativePath" is the path to the asset relative to the root of the package.
 *
 * For example, suppose you have a package called "my-fancy-package".
 * This package expects at runtime to have access to all of the binary files in the folder "<package-root>/resources/binaries"
 * as well as access to a jar file at "<package-root>/resources/jars/myJar.jar"
 *
 * In that case, this package will export the following function:
 *
 * export const getPackageAssetPaths = () => ['resources/binaries', 'resources/jars/myJar.jar'];
 *
 * A binary could then be accessed at path.join(pathManager.getAmplifyPackageLibDirPath('my-fancy-package'), 'resources/binaries', 'myBinary')
 * Likewise the jar can be retrieved at path.join(pathManager.getAmplifyPackageLibDirPath('my-fancy-package'), 'resources/jars/myJar.jar')
 */
export type GetPackageAssetPaths = () => Promise<string[]>;

export type $IPluginManifest = $TSAny;

// Use it for all file content read from amplify-meta.json
export type $TSMeta = $TSAny;

// Use it for all file content read from team-provider-info.json
export type $TSTeamProviderInfo = $TSAny;

// Use it for all object initializer usages: {}
export type $TSObject = Record<string, $TSAny>;

// There are tons of places where we use these two pieces of information to identify a resource
// We can use this type to type those instances
export interface ResourceTuple {
  category: string;
  resourceName: string;
}

export enum AmplifyFrontend {
  android = 'android',
  ios = 'ios',
  javascript = 'javascript',
}
export interface AmplifyProjectConfig {
  projectName: string;
  version: string;
  frontend: AmplifyFrontend;
  providers: string[];
}

/**
 * higher level context object that could be used in plugins
 */
export interface ProviderContext {
  provider: string;
  service: string;
  projectName: string;
}

export type $TSCopyJob = $TSAny;

// Temporary interface until Context refactor
interface AmplifyToolkit {
  confirmPrompt: (prompt: string, defaultValue?: boolean) => Promise<boolean>;
  constants: $TSAny;
  constructExeInfo: (context: $TSContext) => $TSAny;
  copyBatch: (context: $TSContext, jobs: $TSCopyJob[], props: object, force?: boolean, writeParams?: boolean | object) => $TSAny;
  crudFlow: (role: string, permissionMap?: $TSObject, defaults?: string[]) => Promise<string[]>;
  deleteProject: () => $TSAny;
  executeProviderUtils: (context: $TSContext, providerName: string, utilName: string, options?: $TSAny) => Promise<$TSAny>;
  getAllEnvs: () => string[];
  getPlugin: () => $TSAny;
  getCategoryPluginInfo: (context: $TSContext, category?: string, service?: string) => $TSAny;
  getAllCategoryPluginInfo: (context: $TSContext) => $TSAny;
  getFrontendPlugins: (context: $TSContext) => $TSAny;
  getEnvDetails: () => $TSAny;
  getEnvInfo: () => $TSAny;
  getProviderPlugins: (context: $TSContext) => Record<string, string>;
  getPluginInstance: (context: $TSContext, pluginName: string) => $TSAny;
  getProjectConfig: () => $TSAny;
  getProjectDetails: () => $TSAny;

  /**
   * @deprecated Use stateManager.getMeta() from amplify-cli-core
   */
  getProjectMeta: () => $TSMeta;
  getResourceStatus: (category?: $TSAny, resourceName?: $TSAny, providerName?: $TSAny, filteredResources?: $TSAny) => $TSAny;
  getResourceOutputs: () => $TSAny;
  getWhen: () => $TSAny;
  /**
   * @deprecated Use validators from amplify-prompts or add a new validator in that module
   */
  inputValidation: (input: $TSAny) => (value: $TSAny) => boolean | string;
  listCategories: () => $TSAny;

  /**
   * @deprecated use uuid
   */
  makeId: (n?: number) => string;
  openEditor: (context: $TSContext, target: string, waitToContinue?: boolean) => Promise<void>;
  onCategoryOutputsChange: (context: $TSContext, currentAmplifyMeta: $TSMeta | undefined, amplifyMeta?: $TSMeta) => $TSAny;
  pathManager: $TSAny;
  pressEnterToContinue: () => $TSAny;
  pushResources: (
    context: $TSContext,
    category?: string,
    resourceName?: string,
    filteredResources?: { category: string; resourceName: string }[],
    rebuild?: boolean,
  ) => $TSAny;
  storeCurrentCloudBackend: () => $TSAny;

  /**
   * @deprecated use stateManager or JSONUtilities from amplify-cli-core
   */
  readJsonFile: (fileName: string) => $TSAny;
  removeDeploymentSecrets: (context: $TSContext, category: string, resource: string) => void;
  removeResource: (
    context: $TSContext,
    category: string,
    resource: string,
    questionOptions?: {
      headless?: boolean;
      serviceSuffix?: { [serviceName: string]: string };
      serviceDeletionInfo?: { [serviceName: string]: string };
    },
    resourceNameCallback?: (resourceName: string) => Promise<void>,
  ) => Promise<{ service: string; resourceName: string } | undefined>;
  sharedQuestions: () => $TSAny;
  showAllHelp: () => $TSAny;
  showHelp: (header: string, commands: { name: string; description: string }[]) => $TSAny;
  showHelpfulProviderLinks: (context: $TSContext) => $TSAny;
  showResourceTable: (category?: $TSAny, resourceName?: $TSAny, filteredResources?: $TSAny) => Promise<$TSAny>;
  showStatusTable: (resourceTableParams: ViewResourceTableParams) => Promise<$TSAny>; // Enhanced Status with CFN-Diff
  serviceSelectionPrompt: (
    context: $TSContext,
    category: string,
    servicesMetadata: $TSAny,
    customQuestion?: $TSAny,
    optionNameOverrides?: Record<string, string>,
  ) => Promise<ServiceSelection>;
  updateProjectConfig: () => $TSAny;
  updateamplifyMetaAfterResourceUpdate: (
    category: string,
    resourceName: string,
    metaResourceKey: string,
    metaResourceData?: $TSAny,
  ) => $TSMeta;
  updateamplifyMetaAfterResourceAdd: (
    category: string,
    resourceName: string,
    metaResourceData: $TSAny,
    backendResourceData?: $TSAny,
    overwriteObjectIfExists?: boolean,
  ) => void;
  updateamplifyMetaAfterResourceDelete: (category: string, resourceName: string) => void;
  updateProvideramplifyMeta: (providerName: string, options: $TSObject) => void;
  updateamplifyMetaAfterPush: (resources: $TSObject[]) => void;
  // buildType is from amplify-function-plugin-interface but can't be imported here because it would create a circular dependency
  updateamplifyMetaAfterBuild: (resource: ResourceTuple, buildType?: string) => void;
  updateAmplifyMetaAfterPackage: (resource: ResourceTuple, zipFilename: string, hash?: { resourceKey: string; hashValue: string }) => void;
  updateBackendConfigAfterResourceAdd: (category: string, resourceName: string, resourceData: $TSObject) => void;
  updateBackendConfigAfterResourceUpdate: (category: string, resourceName: string, attribute: string, value: $TSAny) => void;
  updateBackendConfigAfterResourceRemove: (category: string, resourceName: string) => void;
  loadEnvResourceParameters: (context: $TSContext, category: string, resourceName: string) => $TSAny;
  saveEnvResourceParameters: (context: $TSContext, category: string, resourceName: string, envSpecificParams?: $TSObject) => void;
  removeResourceParameters: (context: $TSContext, category: string, resource: string) => void;
  triggerFlow: () => $TSAny;
  addTrigger: () => $TSAny;
  updateTrigger: () => $TSAny;
  deleteTrigger: (context: $TSContext, name: string, dir: string) => Promise<void>;
  deleteAllTriggers: (previouslySaved: $TSAny, resourceName: string, targetDir: string, context: $TSContext) => Promise<void>;
  deleteDeselectedTriggers: () => $TSAny;
  dependsOnBlock: (context: $TSContext, dependsOnKeys: string[], service: string) => $TSAny;
  getTriggerMetadata: () => $TSAny;
  getTriggerPermissions: (context: $TSContext, triggers: any, category: string, resourceName: string) => $TSAny;
  getTriggerEnvVariables: () => $TSAny;
  getTriggerEnvInputs: () => $TSAny;
  getUserPoolGroupList: () => $TSAny[];
  forceRemoveResource: (context: $TSContext, categoryName: string, name: string, dir: string) => $TSAny;
  writeObjectAsJson: () => $TSAny;
  hashDir: (dir: string, exclude: string[]) => Promise<string>;
  leaveBreadcrumbs: (category: string, resourceName: string, breadcrumbs: unknown) => void;
  readBreadcrumbs: (category: string, resourceName: string) => $TSAny;
  loadRuntimePlugin: (context: $TSContext, pluginId: string) => Promise<$TSAny>;
  getImportedAuthProperties: (context: $TSContext) => {
    imported: boolean;
    userPoolId?: string;
    authRoleArn?: string;
    authRoleName?: string;
    unauthRoleArn?: string;
    unauthRoleName?: string;
  };
  invokePluginMethod: <T>(context: $TSContext, category: string, service: string | undefined, method: string, args: $TSAny[]) => Promise<T>;
}
