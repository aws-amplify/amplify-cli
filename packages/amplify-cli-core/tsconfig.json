{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "outDir": "./lib",
    "rootDir": "./src",
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
  }
}
