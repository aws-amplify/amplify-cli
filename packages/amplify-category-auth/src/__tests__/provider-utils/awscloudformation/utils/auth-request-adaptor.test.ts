import {
  AddAuthRequest,
  CognitoUserAliasAttributes,
  CognitoUserPoolSigninMethod,
  CognitoUserProperty,
  UpdateAuthRequest,
} from 'amplify-headless-interface';
import {
  getAddAuthRequestAdaptor,
  getUpdateAuthRequestAdaptor,
} from '../../../../provider-utils/awscloudformation/utils/auth-request-adaptors';
import { FeatureFlags } from 'amplify-cli-core';

describe('get add auth request adaptor', () => {
  beforeEach(() => {
    FeatureFlags.getBoolean = () => false;
  });
  describe('valid translations', () => {
    it('translates request with minimal user pool config only', () => {
      const addAuthRequest: AddAuthRequest = {
        version: 1,
        resourceName: 'myTestAuth',
        serviceConfiguration: {
          serviceName: 'Cognito',
          userPoolConfiguration: {
            signinMethod: CognitoUserPoolSigninMethod.EMAIL,
            requiredSignupAttributes: [CognitoUserProperty.EMAIL],
          },
          includeIdentityPool: false,
        },
      };

      expect(getAddAuthRequestAdaptor('javascript')(addAuthRequest)).toMatchSnapshot();
    });
  });
  it('translates request with aliasAttributes', () => {
    FeatureFlags.getBoolean = () => true;
    const addAuthRequest: AddAuthRequest = {
      version: 1,
      resourceName: 'myTestAuth',
      serviceConfiguration: {
        serviceName: 'Cognito',
        userPoolConfiguration: {
          signinMethod: CognitoUserPoolSigninMethod.EMAIL,
          requiredSignupAttributes: [CognitoUserProperty.EMAIL],
          aliasAttributes: [
            CognitoUserAliasAttributes.EMAIL,
            CognitoUserAliasAttributes.PHONE_NUMBER,
            CognitoUserAliasAttributes.PREFERRED_USERNAME,
          ],
        },
        includeIdentityPool: false,
      },
    };

    expect(getAddAuthRequestAdaptor('javascript')(addAuthRequest)).toMatchSnapshot();
  });
  it('translates request without aliasAttributes', () => {
    FeatureFlags.getBoolean = () => true;
    const addAuthRequest: AddAuthRequest = {
      version: 1,
      resourceName: 'myTestAuth',
      serviceConfiguration: {
        serviceName: 'Cognito',
        userPoolConfiguration: {
          signinMethod: CognitoUserPoolSigninMethod.EMAIL,
          requiredSignupAttributes: [CognitoUserProperty.EMAIL],
          aliasAttributes: [
            CognitoUserAliasAttributes.EMAIL,
            CognitoUserAliasAttributes.PHONE_NUMBER,
            CognitoUserAliasAttributes.PREFERRED_USERNAME,
          ],
        },
        includeIdentityPool: false,
      },
    };

    expect(getAddAuthRequestAdaptor('javascript')(addAuthRequest)).toMatchSnapshot();
  });
});

describe('get update auth request adaptor', () => {
  describe('valid translations', () => {
    it('translates empty oAuth config into hostedUI: false', () => {
      const updateAuthRequest: UpdateAuthRequest = {
        version: 1,
        serviceModification: {
          serviceName: 'Cognito',
          userPoolModification: {
            oAuth: {},
          },
          includeIdentityPool: false,
        },
      };

      expect(getUpdateAuthRequestAdaptor('javascript', ['required_attribute'])(updateAuthRequest)).toMatchSnapshot();
    });
  });
});
