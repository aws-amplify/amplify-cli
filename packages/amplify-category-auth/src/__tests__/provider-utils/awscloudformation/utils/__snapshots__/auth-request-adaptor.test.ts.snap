// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get add auth request adaptor translates request with aliasAttributes 1`] = `
Object {
  "adminQueries": false,
  "adminQueryGroup": undefined,
  "aliasAttributes": Array [
    "email",
    "phone_number",
    "preferred_username",
  ],
  "authProviders": Array [],
  "authSelections": "userPoolOnly",
  "autoVerifiedAttributes": Array [],
  "identityPoolName": undefined,
  "mfaConfiguration": "OFF",
  "requiredAttributes": Array [
    "email",
  ],
  "resourceName": "myTestAuth",
  "serviceName": "Cognito",
  "thirdPartyAuth": false,
  "updateFlow": "manual",
  "useDefault": "manual",
  "userPoolGroupList": Array [],
  "userPoolGroups": false,
  "userPoolName": undefined,
  "usernameAttributes": Array [
    "email",
  ],
  "userpoolClientReadAttributes": Array [],
  "userpoolClientRefreshTokenValidity": undefined,
  "userpoolClientWriteAttributes": Array [],
}
`;

exports[`get add auth request adaptor translates request without aliasAttributes 1`] = `
Object {
  "adminQueries": false,
  "adminQueryGroup": undefined,
  "aliasAttributes": Array [
    "email",
    "phone_number",
    "preferred_username",
  ],
  "authProviders": Array [],
  "authSelections": "userPoolOnly",
  "autoVerifiedAttributes": Array [],
  "identityPoolName": undefined,
  "mfaConfiguration": "OFF",
  "requiredAttributes": Array [
    "email",
  ],
  "resourceName": "myTestAuth",
  "serviceName": "Cognito",
  "thirdPartyAuth": false,
  "updateFlow": "manual",
  "useDefault": "manual",
  "userPoolGroupList": Array [],
  "userPoolGroups": false,
  "userPoolName": undefined,
  "usernameAttributes": Array [
    "email",
  ],
  "userpoolClientReadAttributes": Array [],
  "userpoolClientRefreshTokenValidity": undefined,
  "userpoolClientWriteAttributes": Array [],
}
`;

exports[`get add auth request adaptor valid translations translates request with minimal user pool config only 1`] = `
Object {
  "adminQueries": false,
  "adminQueryGroup": undefined,
  "aliasAttributes": Array [],
  "authProviders": Array [],
  "authSelections": "userPoolOnly",
  "autoVerifiedAttributes": Array [],
  "identityPoolName": undefined,
  "mfaConfiguration": "OFF",
  "requiredAttributes": Array [
    "email",
  ],
  "resourceName": "myTestAuth",
  "serviceName": "Cognito",
  "thirdPartyAuth": false,
  "updateFlow": "manual",
  "useDefault": "manual",
  "userPoolGroupList": Array [],
  "userPoolGroups": false,
  "userPoolName": undefined,
  "usernameAttributes": Array [
    "email",
  ],
  "userpoolClientReadAttributes": Array [],
  "userpoolClientRefreshTokenValidity": undefined,
  "userpoolClientWriteAttributes": Array [],
}
`;

exports[`get update auth request adaptor valid translations translates empty oAuth config into hostedUI: false 1`] = `
Object {
  "adminQueries": false,
  "adminQueryGroup": undefined,
  "authProviders": Array [],
  "authSelections": "userPoolOnly",
  "autoVerifiedAttributes": Array [],
  "hostedUI": false,
  "mfaConfiguration": "OFF",
  "requiredAttributes": Array [
    "required_attribute",
  ],
  "serviceName": "Cognito",
  "thirdPartyAuth": false,
  "updateFlow": "manual",
  "useDefault": "manual",
  "userPoolGroupList": Array [],
  "userPoolGroups": false,
  "userpoolClientReadAttributes": Array [],
  "userpoolClientRefreshTokenValidity": undefined,
  "userpoolClientWriteAttributes": Array [],
}
`;
